### PR TITLE
[ci] Passing ray checkout dir location to docker

### PR DESCRIPTION
**Pull request by can-anyscale**
The variable BUILDKITE_HOOK_WORKING_DIR is controlled by buildkite and is not really passable into job environment. Create our own variable name to store and pass the ray checkout directory to inside docker instead. This is needed for the docker running inside docker to mount to the host machine.

This change is required for https://github.com/ray-project/ray/pull/38639/files to work.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]

File: raycicmd/bk_pipeline.go [hunk 1]
```diff
@@ -49,6 +48,7 @@ var buildkiteEnvs = []string{
 	"BUILDKITE_PARALLEL_JOB_COUNT",
 	"BUILDKITE_PULL_REQUEST",
 	"BUILDKITE_MESSAGE",
+	"RAYCI_CHECKOUT_DIR",
```
Comment (aslonnie) [inline raycicmd/bk_pipeline.go:51]: this is not buildkiteEnvs though.

**Comment by can-anyscale** [comment]
Discussed offline, there is probably a more systematic implementation for this. We'll do that as a follow. I'll merge this to unblock me on migrating the wheel + docker build to civ2 for now.

**Comment by aslonnie** [comment]
wait wait.. I know what you want. I will do it for you in another way.. please hold.

**Comment by can-anyscale** [comment]
SGTM!

**Pull request closed without merging**